packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -979,8 +979,13 @@ class ModalBottomSheetRoute<T> extends PopupRoute<T> {
   /// If true, a [SafeArea] is inserted to keep the bottom sheet away from
   /// system intrusions at the top, left, and right sides of the screen.
   ///
-  /// If false, the bottom sheet isn't exposed to the top padding of the
-  /// MediaQuery.
+  /// If false, the bottom sheet will extend through any system intrusions
+  /// at the top, left, and right, and callers may wish to pad those
+  /// with their own [SafeArea] in [builder]. (Such a [SafeArea] will not
+  /// actually provide top padding, because [builder] will be run
+  /// under a [MediaQuery.removePadding] that has `removeTop: true`.
+  /// To work around, wrap the [SafeArea] in a [MediaQuery] that restates an
+  /// ambient [MediaQueryData] from outside [builder].)
```

Comment:
From the discussion (particularly around https://github.com/flutter/flutter/pull/122225#issuecomment-1690729628 ), it seems like the best advice for most developers will be to set `useSafeArea: true`. So we can perhaps foreground that over the workaround of looking at `MediaQuery.of` from an outer `BuildContext`.

I think the fact that the top padding gets suppressed in `MediaQuery` is among the key facts this documentation should make clear. This PR's version does say that, but it's within a parenthetical that's framed as explaining how a "callers may wish to" suggestion would behave, which means that a reader scanning for the pure facts about this widget's behavior may overlook it. The version in main has it framed appropriately, but then the "isn't exposed to" wording doesn't make clear what it's saying. So we should take an unambiguous description like in this PR revision, and move it up the information tree to be placed among the basic facts of what this property does.

Here's a version that does both of those:

```suggestion
  /// If false, the bottom sheet will extend through any system intrusions
  /// at the top, left, and right.
  ///
  /// If false, then moreover [MediaQuery.removePadding] will be used
  /// to remove top padding, so that a [SafeArea] widget inside the bottom
  /// sheet will have no effect at the top edge. If this is undesired, consider
  /// setting [useSafeArea] to true. Alternatively, wrap the [SafeArea] in a
  /// [MediaQuery] that restates an ambient [MediaQueryData] from outside [builder].
```